optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterGenerator.java

```diff
@@ -95,410 +272,249 @@ public RosterGenerator() {
         RosterState rosterState = new RosterState();
-        rosterState.setDraftLength(0);
-        rosterState.setFirstDraftDate(LocalDate.now());
-        rosterState.setPublishLength(lengthInDays);
-        rosterState.setPublishNotice(lengthInDays);
-        rosterState.setRotationLength(7);
+        rosterState.setTenantId(tenantId);
+        int publishNotice = 14;
+        rosterState.setPublishNotice(publishNotice);
+        LocalDate firstDraftDate = LocalDate.now()
+                .with(TemporalAdjusters.next(DayOfWeek.MONDAY))
+                .plusDays(publishNotice);
+        rosterState.setFirstDraftDate(firstDraftDate);
+        rosterState.setPublishLength(7);
+        rosterState.setDraftLength(14);
         rosterState.setUnplannedRotationOffset(0);
-        rosterState.setLastHistoricDate(LocalDate.now());
-        entityManager.persist(tenant);
-        configuration.setTenantId(tenant.getId());
-        rosterState.setTenantId(tenant.getId());
+        rosterState.setRotationLength(7);
+        rosterState.setLastHistoricDate(LocalDate.now().minusDays(1));
         entityManager.persist(rosterState);
-        entityManager.persist(configuration);
-        return tenant.getId();
+        return rosterState;
     }
 
-    private List<Skill> createSkillList(Integer tenantId, int size) {
+    private List<Skill> createSkillList(GeneratorType generatorType, Integer tenantId, int size) {
         List<Skill> skillList = new ArrayList<>(size);
-        skillNameGenerator.predictMaximumSizeAndReset(size);
+        generatorType.skillNameGenerator.predictMaximumSizeAndReset(size);
         for (int i = 0; i < size; i++) {
-            String name = skillNameGenerator.generateNextValue();
+            String name = generatorType.skillNameGenerator.generateNextValue();
             Skill skill = new Skill(tenantId, name);
             entityManager.persist(skill);
             skillList.add(skill);
         }
         return skillList;
     }
 
-    private List<Spot> createSpotList(Integer tenantId, int size, List<Skill> skillList) {
+    private List<Spot> createSpotList(GeneratorType generatorType, Integer tenantId, int size, List<Skill> skillList) {
         List<Spot> spotList = new ArrayList<>(size);
-        spotNameGenerator.predictMaximumSizeAndReset(size);
+        generatorType.spotNameGenerator.predictMaximumSizeAndReset(size);
         for (int i = 0; i < size; i++) {
-            String name = spotNameGenerator.generateNextValue();
-            Spot spot = new Spot(tenantId, name, extractRandomSubListOfLength(skillList, random.nextInt(skillList
-                    .size())).stream().collect(Collectors.toSet()));
+            String name = generatorType.spotNameGenerator.generateNextValue();
+            Set<Skill> requiredSkillSet = new HashSet<>(
+                    extractRandomSubListOfSize(skillList, random.nextInt(skillList.size())));
```

Comment:
There is a possibility this will have size 0; do we want to allow spots to have 0 required skills? (Currently some spots have 0 required skills)

Reply:
Yes, we do.
For security guards, it's common to have spots where the guard doesn't need any extra skills beyond being in the system. We want to test this (to flush out NPE's etc), so we need to have such testdata.